third-party/lz4/CMakeLists.txt

```diff
@@ -1,2 +1,30 @@
-add_definitions(-fPIC)
-add_library(lz4 STATIC src/lib/lz4.c src/lib/lz4.h src/lib/lz4hc.c src/lib/lz4hc.h)
+add_library(lz4 INTERFACE)
+
+find_package(LZ4)  # also checks minimum required version
+
+if (LZ4_FOUND)
+  target_include_directories(lz4 INTERFACE ${LZ4_INCLUDE_DIR})
+  target_link_libraries(lz4 INTERFACE ${LZ4_LIBRARY})
+
+else (LZ4_FOUND)
+  cmake_minimum_required(VERSION 2.8.0)
+  include(ExternalProject)
+
+  set(LZ4_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/lz4-install")
+
+  ExternalProject_Add(
+    lz4Build
+    URL "https://github.com/lz4/lz4/archive/v1.9.2.tar.gz"
+    URL_HASH SHA256=658ba6191fa44c92280d4aa2c271b0f4fbc0e34d249578dd05e50e76d0e5efcc
+    PREFIX ${LZ4_PREFIX}
```

Comment:
I think my previous comment was lost by github - this should be `<INSTALL_DIR>` rather than specified, then fetched with ExternalTarget_get_property

Reply:
Example from one of my personal projects: https://github.com/fredemmott/StreamDeck-CPPSDK/blob/fdd50d9972d92b6486d1aa9b3fbe0a3cd7572d6b/Vendor/fmt.cmake

Uses cmake, but the placeholder works regardless